### ISSUE TITLE
the reprojection error loss

### ISSUE THREAD

**Issue by cv-lab-x** [ueda0319/neddf#52]
Hi, 
thanks for your great work. I have some questions about the loss, since i didn't find the defination all the loss in the paper. 
1. can you post all the loss you use to train and test, also split them?
2. do you train a pretrain model before you using the reprojection error loss to do localization?
3. How can i test localization in your codes?

Looking forward to your reply!Thanks!

**Comment by cv-lab-x** [comment]
@ueda0319 @gatheluck @narusemotoki

**Comment by ueda0319** [comment]
Thank you for your question @cv-lab-x  !

1. In our experiments with NeRF_synthetic_dataset, we used a composite of color error, mask error, and regularization penalty of 1.0:0.05:0.01 for Loss, respectively. The color error is MSE, as in NeRF. And the mask error is BCE, as in NeuS. The regularization penalty is described in the paper.
hyperparametes: https://github.com/ueda0319/neddf/blob/master/config/loss/neddf_loss.yaml
codes: https://github.com/ueda0319/neddf/tree/master/neddf/loss

2. Yes, the experiments in the paper use Localization by reprojection error after creating a trained model.
Note that this is to align the experimental conditions. In the future, we believe it will be easy to operate field learning simultaneously, like SLAM.

3. Sorry, this repository is in the process of redesigning for public release, and the Localization part is not yet open to the public.
Please wait for a little while to complete the implementation...

**Comment by cv-lab-x** [comment]
> Thank you for your question @cv-lab-x !
> 
> 1. In our experiments with NeRF_synthetic_dataset, we used a composite of color error, mask error, and regularization penalty of 1.0:0.05:0.01 for Loss, respectively. The color error is MSE, as in NeRF. And the mask error is BCE, as in NeuS. The regularization penalty is described in the paper.
>    hyperparametes: https://github.com/ueda0319/neddf/blob/master/config/loss/neddf_loss.yaml
>    codes: https://github.com/ueda0319/neddf/tree/master/neddf/loss
> 2. Yes, the experiments in the paper use Localization by reprojection error after creating a trained model.
>    Note that this is to align the experimental conditions. In the future, we believe it will be easy to operate field learning simultaneously, like SLAM.
> 3. Sorry, this repository is in the process of redesigning for public release, and the Localization part is not yet open to the public.
>    Please wait for a little while to complete the implementation...


@ueda0319 
Thanks for your reply, another question, 
1. did you do the experiments about jointing optimization of scene representations and camera poses ?  
2. or Is it feasible to joint optimization of scene representations and camera poses using reprojection error loss and color error and BCE loss ?

**Comment by ueda0319** [comment]
No. In the experiments of this paper, scene representation and camera pose optimization are trained separately.
